vm/src/stdlib/binascii.rs

```diff
@@ -177,31 +177,113 @@ mod decl {
         encoded
     }
 
+    #[inline]
+    fn uu_a2b_read(s: &[u8], index: usize, vm: &VirtualMachine) -> PyResult<u8> {
+        if let Some(c) = s.get(index) {
+            // Check the character for legality
+            // The 64 instead of the expected 63 is because
+            // there are a few uuencodes out there that use
+            // '`' as zero instead of space.
+            if *c < 32 || *c > 96 {
+                if *c == 10 || *c == 13 {
+                    return Ok(0);
+                }
+                return Err(vm.new_value_error("Illegal char".to_string()));
+            }
+            Ok((*c - 0x20) & 0x3f)
+        } else {
+            Ok(0)
+        }
+    }
+
     #[pyfunction]
     fn a2b_uu(s: SerializedData, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
         s.with_ref(|b| {
-            let mut buf;
-            let b = if memchr::memchr(b'\n', b).is_some() {
-                buf = b.to_vec();
-                buf.retain(|c| *c != b'\n');
-                &buf
+            // First byte: binary data length (in bytes)
+            let length = if b.is_empty() {
+                ((-0x20i32) & 0x3fi32) as usize
             } else {
-                b
+                ((b[0] - 0x20) & 0x3f) as usize
             };
-            // TODO: RUSTPYTHON, implement actual uuencoding code
-            base64::decode(b)
+
+            // Allocate the buffer
+            let mut res = Vec::<u8>::with_capacity(length);
+
+            for i in (1..b.len()).step_by(4) {
```

Comment:
Rather than manually manipulating index and chunks, How about this way?
```suggestion
let (chunks, remainder) =  b.as_chunks::<4>();
for (i, chunk) in chunks.enumerate() {
```